### PR TITLE
Adding useHotKeys module for better keyPress event handling

### PR DESCRIPTION
**Pull request by avnkailash**

### CODE REVIEW

**Comment by avnkailash** [comment]
Do not merge it. I want to test the change locally and then merge it. Code-wise changes are all done and complete!

File: src/elements/fields/TextField.jsx [hunk 1]
```diff
@@ -89,6 +90,11 @@ function TextField({
 }) {
   const servar = element.servar;
   const inputType = fieldProps.as === 'textarea' ? 'textarea' : 'input';
+
+  useHotkeys('enter', (e) =>
```
Comment (bo-dun-1) [inline src/elements/fields/TextField.jsx:94]: when is this event triggered? before we would specify the element on which we're listening to keydown but now that it's in a use effect, what happens?

Comment (avnkailash) [reply src/elements/fields/TextField.jsx]: useHotKeys hook will listen for all key interactions on the element. When the text field is in focus, the hook will capture the key events and call the callback function(2nd argument). This technically replaces the onKeyDown event handler.

File: src/elements/fields/PinInputField.jsx [hunk 2]
```diff
@@ -41,6 +34,42 @@ function SingleOtpInput({
...
 
   applyStyles.applyBorders('field', focus ? 'selected_' : '', false);
+
+  // Handle cases of backspace, delete, left arrow, right arrow, space
+  useHotkeys(
+    'enter, backspace, delete, left, right, space',
+    (e, handler) => {
+      switch (handler.key) {
+        case 'enter':
+          console.log('You pressed enter!');
```
Comment (bo-dun-1) [inline src/elements/fields/PinInputField.jsx:44]: this doens't look like final code

File: src/elements/fields/TextField.jsx [hunk 3]
```diff
@@ -89,6 +90,15 @@ function TextField({
 }) {
   const servar = element.servar;
   const inputType = fieldProps.as === 'textarea' ? 'textarea' : 'input';
+
+  useHotkeys(
+    'enter',
+    (e) => (inputType === 'textarea' ? e.stopPropagation() : null),
```
Comment (bo-dun-1) [inline src/elements/fields/TextField.jsx:96]: can this instead be inputType === 'textarea' && e.stopPropagation()

Comment (bo-dun-1) [reply src/elements/fields/TextField.jsx:96]: also have you tested that enter successfully creates newlines for textareas on forms and won't attemp to submit that step? but will submit for text fields?

File: src/form/Form.jsx [hunk 4]
```diff
@@ -1012,6 +1013,16 @@ function Form({
     } else handleRedirect({ metadata });
   };
 
+  useHotkeys('enter', (e) => {
```
Comment (bo-dun-1) [inline src/form/Form.jsx:1016]: have u verified the submit functionality still works as expected?

**Review by bo-dun-1 — CHANGES_REQUESTED** [review]

File: src/elements/fields/PinInputField.jsx (hunk 2 above)
Comment (avnkailash) [reply src/elements/fields/PinInputField.jsx]: Fixed it. Was testing the enter key event and missed changing the line.

File: src/elements/fields/TextField.jsx (hunk 3 above)
Comment (avnkailash) [reply src/elements/fields/TextField.jsx]: Done. Changed the logic to use && operator. Yes I have tested the key event and validated that it behaves as expected in the text area

File: src/form/Form.jsx (hunk 4 above)
Comment (avnkailash) [reply src/form/Form.jsx]: Yes, I have tested. I had to add an option to make it behave correctly and consistently.

File: src/elements/fields/PinInputField.jsx (hunk 2 above)
Comment (avnkailash) [reply src/elements/fields/PinInputField.jsx]: As per my testing of the existing PinInput field, I am not doing any specific action except for preventing the default behavior. Should we change that behavior?

**Comment by avnkailash** [comment]
As per my testing of the existing PinInput field, I am not doing any specific action except for preventing the default behavior. Should we change that behavior?

File: package.json [hunk 5]
```diff
@@ -68,11 +68,13 @@
     "@emotion/styled": "^11.0.0",
     "@fingerprintjs/fingerprintjs": "^3.1.0",
     "google-libphonenumber": "^3.2.22",
+    "imask": "^6.2.2",
```
Comment (bo-dun-1) [inline package.json:71]: why are we adding imask here when we have react-imask?

**Comment by bo-dun-1** [comment]
> As per my testing of the existing PinInput field, I am not doing any specific action except for preventing the default behavior. Should we change that behavior?

You should test out the existing key listeners, e.g. enter to submit steps with only 1 field

**Review by bo-dun-1 — CHANGES_REQUESTED** [review]

File: src/elements/fields/PinInputField.jsx [hunk 6]
```diff
@@ -60,7 +92,7 @@ function SingleOtpInput({
         ref={input}
         value={value || ''}
         onChange={onChange}
-        onKeyDown={onKeyDown}
+        // onKeyDown={onKeyDown}
```
Comment (bo-dun-1) [inline src/elements/fields/PinInputField.jsx:95]: don't leeave commented out code in production code

File: src/elements/fields/TextField.jsx [hunk 7]
```diff
@@ -89,6 +90,15 @@ function TextField({
 }) {
   const servar = element.servar;
   const inputType = fieldProps.as === 'textarea' ? 'textarea' : 'input';
+
+  useHotkeys(
+    'enter',
+    (e) => (inputType === 'textarea' && e.stopPropagation()),
```
Comment (bo-dun-1) [inline src/elements/fields/TextField.jsx:96]: why not just enable this only on inputs and remove this if check? also this is currently buggy, enter will submit text areas even though it shouldn't. did you test this code path?

**Review by bo-dun-1 — CHANGES_REQUESTED** [review]
this pr needs to be rebased

File: package.json (hunk 5 above)
Comment (avnkailash) [reply package.json]: This has been removed.

File: src/elements/fields/PinInputField.jsx (hunk 6 above)
Comment (avnkailash) [reply src/elements/fields/PinInputField.jsx]: done

File: src/elements/fields/TextField.jsx (hunk 7 above)
Comment (avnkailash) [reply src/elements/fields/TextField.jsx]: Modified this logic to capture only the text area. `enter` events and stop the propagation.

**Pull request merged**